test/lognormal.jl

```diff
@@ -11,8 +11,9 @@ isnan_type(::Type{T}, v) where {T} = isnan(v) && v isa T
     @test logpdf(LogNormal(), Inf) === -Inf
     @test iszero(logcdf(LogNormal(0, 0), 1))
     @test iszero(logcdf(LogNormal(), Inf))
-    @test logdiffcdf(LogNormal(), Float32(exp(5)), Float32(exp(3))) ≈ -6.6079385945968929 rtol=1e-12
-    @test logdiffcdf(LogNormal(), Float64(exp(5)), Float64(exp(3))) ≈ -6.6079385945968929 rtol=1e-12
+    @test logdiffcdf(LogNormal(), Float32(exp(5)), Float32(exp(3))) ≈ -6.607938594596893 rtol=1e-12
+    @test logdiffcdf(LogNormal(), Float32(exp(5)), Float64(exp(3))) ≈ -6.60793859457367 rtol=1e-12
+    @test logdiffcdf(LogNormal(), Float64(exp(5)), Float64(exp(3))) ≈ -6.607938594596893 rtol=1e-12
```

Comment:
why did the test values change? did precision change? the old tests pass for me (same with `Normal`) :)

also it's kinda a shame we don't have a test like
```julia
@test @test _logdiffcdf(d, x, y) ≈ log(cdf(d, x) - cdf(d, y))
```

Reply:
I updated the values to the actual results of the computations. Most values did not change (it's just ...3 instead of ...29, but since the REPL only prints ...3 I thought that would be more appropriate), but
```julia
logdiffcdf(LogNormal(), Float32(exp(5)), Float64(exp(3)))
```
yields a more different result since precision is lost when computing `Float32(exp(5))` (even though it's promoted back to `Float64` inside of `logdiffcdf`). This computation was not possible before since `logdiffcdf` only allowed input arguments of the same type. As expected, when evaluating `logdiffcdf(Normal(), Float32(5), Float64(3))` these promotions and conversions don't cause any loss of precision.

Reply:
> since the REPL only prints ...3 I thought that would be more appropriate

cool, that makes sense to me. thanks for explaining! LGTM